internal/step/foreach/provider.go

```diff
@@ -442,33 +443,36 @@ func (r *runnableStep) Start(_ map[string]any, runID string, stageChangeHandler
 		lock:               &sync.Mutex{},
 		currentStage:       StageIDEnabling,
 		currentState:       step.RunningStepStateStarting,
-		inputData:          make(chan []any, 1),
+		executeInput:       make(chan executeInput, 1),
 		enabledInput:       make(chan bool, 1),
 		workflow:           r.workflow,
 		stageChangeHandler: stageChangeHandler,
-		parallelism:        r.parallelism,
 		logger:             r.logger,
 	}
 	go rs.run()
 	return rs, nil
 }
 
+type executeInput struct {
+	inputData   []any
+	parallelism int64
+}
```

Comment:
Are you sure you want a field name (at line 446) with the same name as a type?

Reply:
I dislike Golang's naming of having the internal types be lowerCamelCase.

I think this is okay. It's done the same way elsewhere.